azurelinuxagent/common/protocol/metadata.py

```diff
@@ -314,8 +314,9 @@ def update_goal_state(self, forced=False, max_retry=3):
             try:
                 self.update_certs()
                 return
-            except:
+            except Exception as e:
                 logger.verbose("Incarnation is out of date. Update goalstate.")
+                logger.verbose("Exception: %{0}".format(e))
```

Comment:
Logging the traceback would help as well to identify the issue. Reference - https://github.com/Azure/WALinuxAgent/blob/master/azurelinuxagent/ga/exthandlers.py#L219.

Reply:
@vrdmr updated.  Let me know what you think.  I ended up copying the get_traceback function over... copy and paste not ideal, but I didn't want a crazy import loop either to get it.  I am happy to move it out to a utils method if you'd like?